src/Compilers/CSharp/Portable/Compilation/MemberSemanticModel.cs

```diff
@@ -1788,6 +1809,73 @@ private static Binder GetLambdaEnclosingBinder(int position, CSharpSyntaxNode st
             return GetEnclosingBinderInternalWithinRoot(startingNode, position, lambdaBinder, containingLambda);
         }
 
+        /// <summary>
+        /// If we're doing nullable analysis, we need to fully bind this member, and then run
+        /// nullable analysis on the resulting nodes before putting them in the map. Nullable
+        /// analysis does not run a subset of code, so we need to fully bind the entire member
+        /// first
+        /// </summary>
+        private void EnsureRootBoundForNullabilityIfNecessary()
+        {
+            // In DEBUG without nullable analysis enabled, we want to use a temp diagnosticbag
+            // that can't produce any observable side effects
+            DiagnosticBag diagnostics = _ignoredDiagnostics;
+
+            // If we're in DEBUG mode, always enable the analysis, but throw away the results
+            if (!EnableNullableAnalysis)
+            {
+#if DEBUG
+                diagnostics = new DiagnosticBag();
+#else
+                return;
+#endif
+            }
+
+            using var upgradeableLock = _nodeMapLock.DisposableUpgradeableRead();
+
+            if (_guardedNodeMap.ContainsKey(Root)
+#if DEBUG
+                // In DEBUG mode, we don't want to increase test run times by an insane length of
+                // time, so if nullable analysis isn't enabled and some node has already been bound
+                // we assume we've already done this test binding and just return
+                || (!EnableNullableAnalysis && _guardedNodeMap.Count > 0)
+#endif
+                )
+            {
+#if DEBUG
+                if (!EnableNullableAnalysis) diagnostics.Free();
+#endif
+
+                return;
+            }
+
+            upgradeableLock.EnterWrite();
+
+            Debug.Assert(Root == GetBindableSyntaxNode(Root));
```

Comment:
>GetBindableSyntaxNode(Root) [](start = 33, length = 27)

For some reason the function below explicitly calls ```GetBindableSyntaxNode(Root)``` for production code, I think it would be safer if we do the same in this function and simply operate on the result we get back rather than using ```Root``` explicitly and assume that the nodes are the same.